packages/image_picker/image_picker/lib/image_picker.dart

```diff
@@ -277,25 +269,50 @@ class ImagePicker {
     int? imageQuality,
     bool requestFullMetadata = true,
   }) {
-    if (imageQuality != null && (imageQuality < 0 || imageQuality > 100)) {
-      throw ArgumentError.value(
-          imageQuality, 'imageQuality', 'must be between 0 and 100');
-    }
-    if (maxWidth != null && maxWidth < 0) {
-      throw ArgumentError.value(maxWidth, 'maxWidth', 'cannot be negative');
-    }
-    if (maxHeight != null && maxHeight < 0) {
-      throw ArgumentError.value(maxHeight, 'maxHeight', 'cannot be negative');
-    }
+    final ImageOptions imageOptions = ImageOptions.createAndValidate(
+      maxWidth: maxWidth,
+      maxHeight: maxHeight,
+      imageQuality: imageQuality,
+      requestFullMetadata: requestFullMetadata,
+    );
 
     return platform.getMultiImageWithOptions(
       options: MultiImagePickerOptions(
-        imageOptions: ImageOptions(
-          maxWidth: maxWidth,
-          maxHeight: maxHeight,
-          imageQuality: imageQuality,
-          requestFullMetadata: requestFullMetadata,
-        ),
+        imageOptions: imageOptions,
+      ),
+    );
+  }
+
+  /// Returns a [List<XFile>] with the images and/or videos that were picked.
+  /// The images and videos come from the gallery.
+  ///
+  /// Where iOS supports HEIC images, Android 8 and below doesn't. Android 9 and
+  /// above only support HEIC images if used in addition to a size modification,
+  /// of which the usage is explained below.
+  ///
+  /// This method is only supported on Android API 19 and above.
+  /// It could possibly work on API 18 and below on some devices,
+  /// but this is not guaranteed.
+  ///
+  /// In Android, the MainActivity can be destroyed for various reasons.
+  /// If that happens, the result will be lost in this call. You can then
+  /// call [getLostData] when your app relaunches to retrieve the lost data.
+  ///
+  /// If no images or videos were picked, the return value is an empty list.
+  Future<List<XFile>> pickMedia({
```

Comment:
Do we have easy access to the type (image or video) information on the native side when we get a response? If so, it would be nice to preserve it in the return, by making this a `List<SomeNewDataClass>` that has an XFile and an enum for the media type.

That would make the example less hacky, for instance, by knowing which viewer to use for each entry.

Reply:
Do you think giving the full mime type as a string would be better? I'm open to either, but mime type gives more info